srcs/client/src/Pages/Game/Game.tsx

```diff
@@ -0,0 +1,115 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import Canvas from '../../Components/Canvas/Canvas';
+
+const Game = (props: any) => {
+  // Enum
+  enum movement {
+    UP = 0,
+    DOWN,
+  }
+
+  const enum Motive {
+    WIN = 0,
+    LOSE,
+    ABANDON,
+    CANCEL,
+  }
+
+  const socket = props.socket;
+  // States
+  const [grid, setGrid] = useState({} as any);
+  const [scores, setScores] = useState([]);
+  const [message, setMessage] = useState('');
+  const timer = useRef({} as NodeJS.Timer);
+
+  // Init
+  const initGame = () => {
+    if (props.action === props.actionVal.JOIN_GAME) {
+      socket.emit('joinGame', { id: props.id });
```

Comment:
I've also created an enum for Events in order to avoid writing magic strings which can be a pain when debugging.

`export enum eEvents {
      CreateSomething: 'CreateSomething,
`

wdyt ?

Reply:
It is indeed more clear, I vote for it ! Next PR